Services/LDAP/classes/class.ilLDAPRoleAssignmentRule.php

```diff
@@ -135,11 +136,14 @@ protected function wildcardCompare(string $a_str1, string $a_str2): bool
 
             try {
                 return preg_match($delimiter . "^" . $pattern . '$' . $delimiter . 'i', $a_str2) === 1;
-            } catch (Exception $ex) {
-                $this->logger->warning('Error occurred in preg_match Ex.: ' . $ex->getMessage());
-            } finally {
-                restore_error_handler();
+            } catch (Exception $e) {
+                $exception = $e;
             }
+            restore_error_handler();
+        }
+
+        if ($exception !== null) {
+            $this->logger->warning('Error occurred in preg_match Ex.: ' . $exception->getMessage());
```

Comment:
I (again) thought about this: Could you please put all exceptions to a stack and then report all errors here.

`Trying the RegEx delimiter chain results in the following problems: %s, %s, %s ....`, where each `%s´ should contain the concrete RegEx and the exception message.